uniflow/op/model/layout_utils.py

```diff
@@ -0,0 +1,348 @@
+
+    _indices = boxes[:, 1].argsort()
+    y_sorted_boxes = boxes[_indices]
+    y_sorted_indices = indices[_indices]
+
+    # debug_vis(y_sorted_boxes, y_sorted_indices)
+
+    y_projection = projection_by_bboxes(boxes=y_sorted_boxes, axis=1)
+    pos_y = split_projection_profile(y_projection, 0, 1)
+    if not pos_y:
+        return
+
+    arr_y0, arr_y1 = pos_y
+    for r0, r1 in zip(arr_y0, arr_y1):
+        _indices = (r0 <= y_sorted_boxes[:, 1]) & (y_sorted_boxes[:, 1] < r1)
+
+        y_sorted_boxes_chunk = y_sorted_boxes[_indices]
+        y_sorted_indices_chunk = y_sorted_indices[_indices]
+
+        _indices = y_sorted_boxes_chunk[:, 0].argsort()
+        x_sorted_boxes_chunk = y_sorted_boxes_chunk[_indices]
+        x_sorted_indices_chunk = y_sorted_indices_chunk[_indices]
+
+        x_projection = projection_by_bboxes(boxes=x_sorted_boxes_chunk, axis=0)
+        pos_x = split_projection_profile(x_projection, 0, 1)
+        if not pos_x:
+            continue
+
+        arr_x0, arr_x1 = pos_x
+        if len(arr_x0) == 1:
+            res.extend(x_sorted_indices_chunk)
+            continue
+
+        for c0, c1 in zip(arr_x0, arr_x1):
+            _indices = (c0 <= x_sorted_boxes_chunk[:, 0]) & (
+                x_sorted_boxes_chunk[:, 0] < c1
+            )
+            recursive_xy_cut(
+                x_sorted_boxes_chunk[_indices], x_sorted_indices_chunk[_indices], res
+            )
+
+
+class LayoutPredictor:
+    """
+    Class for predicting layout of an image.
+
+    Attributes:
+        ort_session (onnxruntime.InferenceSession): Inference session for running the model.
+        categorys (dict): Mapping of category index to category name.
+    """
+
+    def __init__(self, model_name: str, model_file: str):
+        """
+        Initializes the LayoutPredictor class.
+        Downloads the model specified by `model_name` and `model_file` from the Hugging Face Model Hub.
+        Creates an inference session using the downloaded model.
+        """
+        model_path = hf_hub_download(model_name, model_file)
+        self.ort_session = onnxruntime.InferenceSession(
+            model_path, providers=["CUDAExecutionProvider", "CPUExecutionProvider"]
+        )
+        self.categorys = {
+            0: "Caption",
+            1: "Footnote",
+            2: "Formula",
+            3: "List-item",
+            4: "Page-footer",
+            5: "Page-header",
+            6: "Picture",
+            7: "Section-header",
+            8: "Table",
+            9: "Text",
+            10: "Title",
+        }
+
+    def __call__(self, img):
+        """
+        Predicts the layout of the input image.
+
+        Args:
+            img (numpy.ndarray): Input image.
+
+        Returns:
+            list: List of dictionaries containing the predicted layout information.
+                  Each dictionary contains the type of the element and its bounding box coordinates.
+        """
+        h, w = (1024, 768)
```

Comment:
Let's move this out of the __call__ and use `H`, `W` to indicate this is class level member variable to improve readability.